app/filters.php

```diff
@@ -179,3 +179,52 @@
     }
   }
 });
+
+Route::filter('isOwner', function($route, $request)
+{
+  $id = Auth::id();
+
+  //get requested id
+  if($route->parameterNames('0')['0'] == 'id')
+  {
+    $requested_id = $route->parameters('id')['id'];
+  }
+  elseif($route->parameterNames('0')['0'] == 'profile')
+  {
+    $requested_id = $route->parameters('profile')['profile'];
+  }
+  elseif($route->parameterNames('0')['0'] == 'application')
+  {
+    $requested_id = $route->parameters('application')['application'];
+  }
+
+  if(!($id == $requested_id))
+  {
+    return Redirect::home()->with('flash_message', ['text' => 'You are not authorized to access that page!', 'class' => '-warning']);
+  }
+});
+
+Route::filter('isOwnerOrAdmin', function($route, $request)
+{
+  $id = Auth::id();
+
+  //get requested id
+  if($route->parameterNames('0')['0'] == 'id')
+  {
+    $requested_id = $route->parameters('id')['id'];
+  }
+  elseif($route->parameterNames('0')['0'] == 'profile')
+  {
+    $requested_id = $route->parameters('profile')['profile'];
+  }
+  elseif($route->parameterNames('0')['0'] == 'application')
+  {
+    $requested_id = $route->parameters('application')['application'];
+  }
```

Comment:
I don't think this works because we can't access $route->parameters (or parameterNames) but we can access $route->parametersNames('0')['0']